### PR TITLE
ddl: wait for a while when some error occured to avoid retrying too many times.

### PR DESCRIPTION
**Pull request by winkyao**
When some error occurred, like tikv is busy or something else, if we do not wait for a while, DDL job will retry immediately. It will act like a deadlock.

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -393,6 +393,11 @@ func (d *ddl) runDDLJob(t *meta.Meta, job *model.Job) (ver int64, err error) {
 
 		job.Error = toTError(err)
 		job.ErrorCount++
+
+		metrics.DDLJobErrCounter.WithLabelValues("failed").Inc()
```
Comment (ciscoxll) [inline ddl/ddl_worker.go]: If there is only one job error, there may be a repeat count.

Comment (winkyao) [reply ddl/ddl_worker.go]: What do you mean?

Comment (ciscoxll) [reply ddl/ddl_worker.go]: The counter has been increasing. `metrics` easy to spot error job.

Comment (coocood) [inline ddl/ddl_worker.go]: Why do we need a label if we only use one label?

**Comment by winkyao** [comment]
@zimulala PTAL

**Comment by winkyao** [comment]
@coocood PTAL

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -393,6 +393,11 @@ func (d *ddl) runDDLJob(t *meta.Meta, job *model.Job) (ver int64, err error) {
 
 		job.Error = toTError(err)
 		job.ErrorCount++
+
+		metrics.DDLJobErrCounter.WithLabelValues(metrics.LabelDDL).Inc()
```
Comment (shenli) [inline ddl/ddl_worker.go]: There is already a metric [here](https://github.com/pingcap/tidb/blob/master/ddl/ddl.go#L458).

File: ddl/syncer.go [hunk 3]
```diff
@@ -37,13 +37,14 @@ const (
...
-	keyOpRetryInterval   = 30 * time.Millisecond
-	checkVersInterval    = 20 * time.Millisecond
+	InitialVersion           = "0"
+	putKeyNoRetry            = 1
+	keyOpDefaultRetryCnt     = 3
+	putKeyRetryUnlimited     = math.MaxInt64
+	keyOpDefaultTimeout      = 2 * time.Second
+	keyOpRetryInterval       = 30 * time.Millisecond
+	checkVersInterval        = 20 * time.Millisecond
+	waitTimeWhenErrorOccured = 50 * time.Millisecond
```
Comment (shenli) [inline ddl/syncer.go]: 50ms is too small. I prefer 1s.

File: ddl/ddl_worker.go (hunk 2 above)
Comment (zimulala) [reply ddl/ddl_worker.go]: There are different metrics. Here collects the count of errors in ddl_worker.go that may not return the error in https://github.com/pingcap/tidb/blob/master/ddl/ddl.go#L458.

**Comment by shenli** [comment]
@winkyao Please address the comments.

**Comment by winkyao** [comment]
@shenli @coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**